eng/doc/fips/UserGuide.md

```diff
@@ -645,6 +658,204 @@ sha512.Sum512_224 is not FIPS compliant.
+
+#### func [VerifyPSS](https://pkg.go.dev/crypto/rsa#VerifyPSS)
+
+```go
+func rsa.VerifyPSS(pub *rsa.PublicKey, hash crypto.Hash, digest []byte, sig []byte, opts *PSSOptions) error
+```
+
+VerifyPSS verifies a PSS signature.
+
+**Parameters**
+
+`hash` can be one of the following values: crypto.MD5, crypto.MD5SHA1, crypto.SHA1, crypto.SHA224, crypto.SHA256, rypto.SHA384, or crypto.SHA512. Else VerifyPSS will fail.
+
+`hashed` must be the result of hashing a message using a FIPS compliant hashing algorithm. If this invariant is not met, VerifyPSS won't be FIPS compliant but still will sign the message.
+
+**Return values**
+
+An error if the signature can't be verified using [EVP_PKEY_verify] with `RSA_PKCS1_PSS_PADDING`.
+
+#### func [GenerateKey](https://pkg.go.dev/crypto/rsa#GenerateKey)
+
+```go
+func rsa.GenerateKey(rand io.Reader, bits int) (priv *rsa.PrivateKey, err error)
+```
+
+GenerateKey generates a public and private key pair.
+
+**Parameters**
+
+`rand` must be boring.RandReader. `crypto/rand.Reader` normally meet this invariant as it is assigned to boring.RandReader in the crypto/rand init function.
+
+`bits` must be either 2048 or 3072.
+
+If any invariant is not met, GenerateMultiPrimeKey won't be FIPS compliant but still will generate the key pair.
+
+**Return values**
+
+`priv` is a wrapper around [EVP_PKEY].
+
+#### func [GenerateMultiPrimeKey](https://pkg.go.dev/crypto/rsa#GenerateMultiPrimeKey)
+
+```go
+func rsa.GenerateMultiPrimeKey(rand io.Reader, nprimes int, bits int) (priv *rsa.PrivateKey, err error)
+```
+
+GenerateMultiPrimeKey generates a multi-prime RSA keypair of the given bit size.
+
+**Parameters**
+
+`rand` must be boring.RandReader. `crypto/rand.Reader` normally meet this invariant as it is assigned to boring.RandReader in the crypto/rand init function.
+
+`nprimes` must be 3. 
+
+`bits` must be either 2048 or 3072.
+
+If any invariant is not met, GenerateMultiPrimeKey won't be FIPS compliant but still will generate the key pair.
+
+**Return values**
+
+`priv` is a wrapper around [EVP_PKEY].
+
+#### func [PrivateKey.Decrypt](https://pkg.go.dev/crypto/rsa#PrivateKey.Decrypt)
+
+```go
+func (priv *PrivateKey) Decrypt(rand io.Reader, ciphertext []byte, opts crypto.DecrypterOpts) (plaintext []byte, err error)
+```
+
+Decrypt decrypts `ciphertext` with `priv`.
+
+If `opts` is nil, `priv.Decrypt` is calls `rsa.DecryptPKCS1v15(rand, priv, ciphertext)`.
+If `opts` is of type `*rsa.OAEPOptions`, `priv.Decrypt` calls `rsa.DecryptOAEP(opts.Hash.New(), rand, priv, ciphertext, opts.Label)`.
+If `opts` is of type `*rsa.PKCS1v15DecryptOptions` and `opts.SessionKeyLen > 0`, `priv.Decrypt` calls `rsa.DecryptPKCS1v15SessionKey(rand, priv, ciphertext, plaintext)` with a random `plaintext`.
+If `opts` is of type `*rsa.PKCS1v15DecryptOptions` and `opts.SessionKeyLen == 0`, `priv.Decrypt` calls `rsa.DecryptPKCS1v15(rand, priv, ciphertext)`.
+Else it returns an error.
+Check those function for the parameters restrictions.

```

Comment:
This looks kind of messy in rendered form without the linebreaks. Word wrap would still potentially make it hard to read even with forced line ends. Maybe use a bulleted list to direct the eye, and use the opportunity to nest the PKCS1v15DecryptOptions options? (Also, suggestion on rewording last sentence.)

```
- If `opts` is nil, `priv.Decrypt` calls `rsa.DecryptPKCS1v15(rand, priv, ciphertext)`.
- If `opts` is of type `*rsa.OAEPOptions`, `priv.Decrypt` calls `rsa.DecryptOAEP(opts.Hash.New(), rand, priv, ciphertext, opts.Label)`.
- If `opts` is of type `*rsa.PKCS1v15DecryptOptions`:
  - If `opts.SessionKeyLen > 0`, `priv.Decrypt` calls `rsa.DecryptPKCS1v15SessionKey(rand, priv, ciphertext, plaintext)` with a random `plaintext`.
  - If `opts.SessionKeyLen == 0`, `priv.Decrypt` calls `rsa.DecryptPKCS1v15(rand, priv, ciphertext)`.
- Else it returns an error.

Check the function named in the applicable case for the parameters' restrictions.
```

Could also add section/anchor links to each func for easier navigation.

(Wanted to mention that I did see https://rhodesmill.org/brandon/2012/one-sentence-per-line/ linked from https://tip.golang.org/doc/comment recently and I do like the idea of semantic linefeeds in general, which might have been the idea here--I just think this bit probably deserves a proper list.)